main.cc

```diff
@@ -685,6 +685,17 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
 #endif
     app_cfg.auto_handle_sigint_sigterm = false;
     app_cfg.reactor_opts.max_networking_io_control_blocks.set_default_value(50000);
+    {
+        const auto candidates = app_cfg.reactor_opts.reactor_backend.get_candidate_names();
+
+        // We don't wan't ScyllaDB to run with the io_uring backend.
+        // So select the default reactor backend explicitely here.
+        if (std::ranges::find(candidates, "linux-aio") == candidates.end()) {
+            app_cfg.reactor_opts.reactor_backend.select_default_candidate("epoll");
+        } else {
+            app_cfg.reactor_opts.reactor_backend.select_default_candidate("linux-aio");
+        }
+    }
```

Comment:
What happens if someone overrides with --reactor-backend=blah? Is it respected?

Reply:
Yes, we only set the default here. The user can override via the command-line.